api/apps/geoprocessing/src/import/pieces-importers/scenario-metadata.piece-importer.ts

```diff
@@ -68,20 +76,42 @@ export class ScenarioMetadataPieceImporter implements ImportPieceProcessor {
       stringScenarioMetadataOrError.right,
     );
 
-    await this.entityManager
-      .createQueryBuilder()
-      .insert()
-      .into('scenarios')
-      .values({
-        id: scenarioId,
-        name,
-        description,
-        blm,
-        number_of_runs: numberOfRuns,
-        metadata,
-        project_id: projectId,
-      })
-      .execute();
+    await this.entityManager.transaction(async (em) => {
+      await em
+        .createQueryBuilder()
+        .insert()
+        .into('scenarios')
+        .values({
+          id: scenarioId,
+          name,
+          description,
+          blm,
+          number_of_runs: numberOfRuns,
+          metadata,
+          project_id: projectId,
+        })
+        .execute();
+
+      if (resourceKind === ResourceKind.Scenario) {
+        await em
+          .createQueryBuilder()
+          .insert()
+          .into(`users_scenarios`)
+          .values({
+            user_id: ownerId,
+            scenario_id: scenarioId,
+            // It would be great to use ScenarioRoles enum instead of having
+            // the role hardcoded. The thing is that Geoprocessing code shouldn't depend
+            // directly on elements of Api code, so there were two options:
+            // - Move ScenarioRoles enum to libs package
+            // - Harcode the rol
+            // We took the second approach because we are only referencing values from that enum
+            // here
+            role_id: 'scenario_owner',
+          })
+          .execute();
```

Comment:
on second thought, you are right (as always 😅) on this - we could skip it and the project owner would get an implicit `scenario_owner` role by the time any scenarios are created (by importing them, in this case) within the project, but the behaviour you implemented here directly maps to the one we have in place when creating scenarios "manually", where the user who creates the scenario is made explicitly a "scenario_owner".